### DISCUSSION TITLE
How do i turn off specific auto analysis in headless? I need mangled name of a function

### DISCUSSION THREAD

**Discussion by anzosasuke** [NationalSecurityAgency/ghidra#5848] [Q&A] [answered]
Hello,

I am looking to print the mangled name of the function and turn off specific Demangler GNU analysis. I looked the DemanglerUtil and DemanglerObject(), but no luck to get the getMangledString() to work. Any help on this is greatly appreciated.

**Comment by ryanmkurtz** [comment, marked as answer]
From a `GhidraScript` passed into headless with `-prescript`:

```java
setAnalysisOption(currentProgram, "Demangler GNU", "false");
```

I did not test this, but used this existing script as an example:

https://github.com/NationalSecurityAgency/ghidra/blob/54e0ab1a3897dad020275d6395acfe0c56c3994d/Ghidra/Features/Base/ghidra_scripts/TurnOffStackAnalysis.java#L24-L34

**Reply by anzosasuke** [reply]
Thanks so much. I used your java version. i was not able to replicate it in python. Any idea why?
Also, getMangledString() function in Demangled class, how would one use it?

**Reply by ryanmkurtz** [reply]
>  i was not able to replicate it in python. Any idea why?

Not sure why it didn't work in python for you...i'd have to see your script.

> Also, getMangledString() function in Demangled class, how would one use it?

As far as I know, the `Demangled` interface is only for use by our internal demanglers.  You would have to kick off demangling yourself (instead of using the analyzer) to get a hold of one of those objects.

**Reply by anzosasuke** [reply]
Thanks so much.
